src/NUnitEngine/nunit.engine/Runners/DirectTestRunner.cs

```diff
@@ -86,10 +98,19 @@ protected override TestEngineResult LoadPackage()
             foreach (var subPackage in packages)
             {
                 var testFile = subPackage.FullName;
+
+                if (_assemblyResolver != null && !TestDomain.IsDefaultAppDomain()
+                    && subPackage.GetSetting(PackageSettings.ImageRequiresDefaultAppDomainAssemblyResolver, false))
+                {
+                    _assemblyResolver.AddPathFromFile(testFile);
+                }
+
                 IFrameworkDriver driver = driverService.GetDriver(TestDomain, testFile);
                 driver.ID = TestPackage.ID;
                 result.Add(driver.Load(testFile, subPackage.Settings));
                 _drivers.Add(driver);
+
+                _packageIndex.Add(driver, subPackage);
```

Comment:
I should clarify on my previous question -- having each user track which entries it added and then remove those entries would be in error if multiple users can submit the same paths. Minor complication, the resolver would have to either track how many adds and removes were issued for each entry or it could allow duplicated entries.